src/sentry/utils/batching_kafka_consumer.py

```diff
@@ -215,6 +215,7 @@ def __init__(
         metrics_sample_rates=None,
         metrics_default_tags=None,
         commit_on_shutdown: bool = False,
+        **_kwargs,
```

Comment:
Please do not do this. We had an issue where we silently ignored CLI arguments because of "convenience" changes like this, and it wasted a bunch of time when we tried to tune those CLI parameters that had no effect, and it took multiple days to fix this for all consumers.

Reply:
This was already brought up before and dismissed by the author: https://github.com/getsentry/sentry/pull/49957#discussion_r1214167717

It seems the intent is compatibility rather than convenience and this will be removed when the migration is complete -- would you prefer we list all options and selectively pass them instead?

Reply:
I did read the response on that comment. I am concerned about this making us silently (and accidentally) ignore CLI parameters that were supposed to be honored, because that's exactly what had already happened in this past (I think it was literally this function that already silently discarded kwargs), and for the exact same stated reason.

If that were only affecting the legacy ingest consumer, I wouldn't care since it will probably be deleted in the next weeks. But batching-kafka-consumer is used by more consumers than that.

I would instead explicitly `options.pop(..)` all unsupported kwargs in `runner/commands/run.py` around line 572. While this still leaves over a bunch of ignored CLI parameters (block size), CLI declaration and list of ignored options are now at least in the same file.

I care about this because this quarter we have an initiative planned (not started) where we attempt to unify all this CLI boilerplate + metrics across consumers, so they work a little bit more like Snuba consumers in their operation.